smali/MagiaNative/app/src/main/java/io/kamihama/magianative/OptionsAllowResponse.java

```diff
@@ -0,0 +1,163 @@
+package io.kamihama.magianative;
+
+import android.content.Context;
+import android.os.Build;
+import android.util.Log;
+import android.webkit.WebResourceRequest;
+import android.webkit.WebResourceResponse;
+import android.webkit.WebView;
+
+import androidx.annotation.RequiresApi;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.math.BigInteger;
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Locale;
+import java.util.Map;
+
+import okhttp3.Headers;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+// https://stackoverflow.com/questions/17272612/android-webview-disable-cors
+public class OptionsAllowResponse {
+    static final String proxyHost = "magirecojp.cirno.name";
+    static final String host = "android.magi-reco.com";
+    static final String topPageJsPath = "/magica/js/top/TopPage.js";
+
+    @RequiresApi(api = Build.VERSION_CODES.LOLLIPOP)
+    private static Request buildRequest(WebResourceRequest request, String mediaType) {
+        // wipe out traces of cross-site request
+        Request.Builder builder = new Request.Builder()
+                .url(request.getUrl().toString());
+
+        if (request.getMethod().equalsIgnoreCase("POST")) {
+            builder = builder.post(RequestBody.create(MediaType.parse(mediaType), "null"));
+        } else {
+            builder = builder.get();
+        }
+
+        builder = builder.removeHeader("Accept-Encoding")
+                .addHeader("Accept-Encoding", "gzip, deflate")
+                .removeHeader("Accept-Language")
+                .addHeader("Accept-Language", "ja-JP,ja;q=0.9,en-US;q=0.8,en;q=0.7")

```

Comment:
Luckily these logcat lines are still present in my terminal. Therefore I can prove that I did not "overwrite" this header without valid reason. Actually I "set" it because I didn't see anything better I can do.

I didn't find `Accept-Language` showing up in overridden `shouldInterceptRequest()` method. It seems that the webview keeps it as its own secret 😂

```
06-20 00:35:46.936 13887 13927 D MagiaHook: LOG MESSAGE: 0.00 0.00
06-20 00:35:46.936 13887 13927 D MagiaHook: Difference: 0x5d0e38
06-20 00:35:53.568 13887 14019 D MagiaHook: Origin: https://android.magi-reco.com
06-20 00:35:53.568 13887 14019 D MagiaHook: User-Agent: Mozilla/5.0 (Linux; Android 12; NCO-AL00 Build/V417IR; wv) AppleWebKit/537.36 (KHTML, like Gecko) Version/4.0 Chrome/91.0.4472.114 Mobile Safari/537.36
06-20 00:35:53.568 13887 14019 D MagiaHook: Referer: https://android.magi-reco.com/
06-20 00:35:53.568 13887 14019 D MagiaHook: f4s-client-ver: 3.1.9
06-20 00:35:53.568 13887 14019 D MagiaHook: webview-session-id: 20240620013550
06-20 00:35:53.568 13887 14019 D MagiaHook: x-platform-host: android.magi-reco.com
06-20 00:35:53.568 13887 14019 D MagiaHook: accept: application/json, text/javascript, */*; q=0.01
06-20 00:35:53.568 13887 14019 D MagiaHook: client-session-id: 33
06-20 00:35:53.568 13887 14019 D MagiaHook: user-id-fba9x88mae: b99d45dd-c92b-4200-8881-8ca4f96444ec
06-20 00:35:53.568 13887 14019 D MagiaHook: x-requested-with: XMLHttpRequest
06-20 00:35:53.568 13887 14019 D MagiaHook: content-type: application/JSON
06-20 00:35:53.568 13887 14019 D MagiaHook: client-model-name: NCO-AL00
06-20 00:35:53.568 13887 14019 D MagiaHook: client-os-ver: 12
```

Reply:
It seems that only after `shouldInterceptRequest()` callback returned, the webview would make its own request, appending the `Accept-Language` header with the value it thinks should be.